src/server/auth/helpers.ts

```diff
@@ -18,9 +18,11 @@ export const verifyCallback = async (
 	})) || { userType: null };
 
 	try {
-		const { emails: [{ value: email }] = [{ value: null }] } = profile;
+		let { emails: [{ value: email }] = [{ value: null }] } = profile;
 		if (email == null) {
-			throw new Error(`Email not provided by provider ${profile}`);
+			// microsoft is fucking stupid, I can't even.
+			email = profile._json.email; // eslint-disable-line
+			if (email == null) throw new Error(`Email not provided by provider ${profile}`);
```

Comment:
LMAO

But please remove profanity from code. Also should rebase because new logging code added.